tests/test_kubernetes_tools.py

```diff
@@ -1316,6 +1317,37 @@ def test_get_boto_volume(self, config_dict):
         else:
             assert volumes is None
 
+    @pytest.mark.parametrize(
+        "config_dict",
+        [
+            {"crypto_keys": {"encrypt": ["mad"], "decrypt": ["max"]}},
+            {"crypto_keys": {"decrypt": ["furiosa"]}},
+            {},
+        ],
+    )
+    def test_get_crypto_volume(self, config_dict):
+        deployment = KubernetesDeploymentConfig(
+            service="my-service",
+            instance="my-instance",
+            cluster="mega-cluster",
+            config_dict=config_dict,
+            branch_dict=None,
+            soa_dir="/nail/blah",
+        )
+        with mock.patch.object(
+            deployment, "get_crypto_secret_hash", return_value="hash"
+        ):
+            volumes = deployment.get_crypto_volume()
+        if config_dict:
+            (volumes.secret.secret_name == "paasta-crypto-key-my-service-my-instance")
+
+            assert len(volumes.secret.items) == len(config_dict["crypto_keys"])
+            assert set(get_crypto_keys_from_config(config_dict["crypto_keys"])) == {
+                item.path.rstrip(".json") for item in volumes.secret.items
+            }
+        else:
+            assert volumes is None
```

Comment:
should we split this test in two - i see that we followed this pattern for the boto tests, but it always makes me a little nervous to have branches in tests :)